ouroboros-network/src/Ouroboros/Network/Pipe.hs

```diff
@@ -25,61 +29,73 @@ data PipeCtx = PipeCtx {
     , pcWrite :: Handle
     }
 
-setupMux :: (Mx.ProtocolEnum ptcl, Ord ptcl, Enum ptcl, Bounded ptcl)
-         => Mx.MiniProtocolDescriptions ptcl IO
-         -> PipeCtx
-         -> IO ()
-setupMux mpds ctx =
-    void $ Mx.muxStart mpds (writePipe ctx) (readPipe ctx) (sduSize ctx) (closePipe ctx) Nothing
+pipeAsMuxBearer
+  :: forall ptcl.
+     ( Mx.ProtocolEnum ptcl
+     , Ord ptcl
+     , Enum ptcl
+     , Bounded ptcl
+     )
+  => PipeCtx
```

Comment:
Does this need it's own type now? This could be just two `Handles` right? Or we could make the pipe directly in this function. I suspect we don't need the `PipeCtx` type at all.

Reply:
In the future I'd like to keep performance/debug counters in SocketCtx and PipeCtx, but perhaps we can keep them some where else.
Anyway there should be a way to monitor what is going on with the node without resorting to wireshark.